eureka2-ext/eureka2-eureka1x-rest-api/src/main/java/com/netflix/eureka2/eureka1x/rest/registry/Eureka1xRegistryProxyImpl.java

```diff
@@ -0,0 +1,123 @@
+    }
+
+    @Override
+    public void register(InstanceInfo instanceInfo) {
+        synchronized (handlers) {
+            RegistrationHandler handler = handlers.get(instanceInfo.getId());
+            if (handler == null) {
+                handler = new RegistrationHandler(registrationClient, scheduler);
+                handlers.put(instanceInfo.getId(), handler);
+            }
+            handler.register(instanceInfo);
+            expiryQueue.enqueue(handler);
+        }
+    }
+
+    @Override
+    public Result appendMeta(String appName, String instanceId, Map<String, String> meta) {
+        synchronized (handlers) {
+            RegistrationHandler handler = handlers.get(instanceId);
+            if (handler != null) {
+                InstanceInfo v1InstanceInfo = handler.getV1InstanceInfo();
+                if (appName.equalsIgnoreCase(v1InstanceInfo.getAppName())) {
+                    InstanceInfo newInstanceInfo = merge(v1InstanceInfo, meta);
+                    handler.register(newInstanceInfo);
+                    return Result.Ok;
+                }
+                return Result.InvalidArguments;
+            }
+        }
+        return Result.NotFound;
+    }
+
+    @Override
+    public Result unregister(String appName, String instanceId) {
+        synchronized (handlers) {
+            RegistrationHandler handler = handlers.remove(instanceId);
+            if (handler != null) {
+                InstanceInfo v1InstanceInfo = handler.getV1InstanceInfo();
+                if (appName.equalsIgnoreCase(v1InstanceInfo.getAppName())) {
+                    handler.unregister();
+                    return Result.Ok;
+                }
+                return Result.InvalidArguments;
+            }
+        }
+        return Result.NotFound;
+    }
+
+    @Override
+    public void unregister(RegistrationHandler cleanupHandler, long expiryTime) {
+        synchronized (handlers) {
+            String instanceId = cleanupHandler.getV1InstanceInfo().getId();
+            RegistrationHandler activeHandler = handlers.get(instanceId);
+            if (cleanupHandler == activeHandler && activeHandler.getExpiryTime() <= expiryTime) {
+                activeHandler.unregister();
+                handlers.remove(instanceId);
+            }
+        }
+    }
+
+    @Override
+    public Result renewLease(String appName, String instanceId) {
+        synchronized (handlers) {
+            RegistrationHandler handler = handlers.get(instanceId);
+            if (handler != null) {
+                InstanceInfo v1InstanceInfo = handler.getV1InstanceInfo();
+                if (appName.equalsIgnoreCase(v1InstanceInfo.getAppName())) {
+                    handler.renew();
+                    expiryQueue.enqueue(handler);
+                    return Result.Ok;
+                }
+                logger.warn("Application name in the request, does not match the one in the instance info ({} != {}, instanceId={}",
```

Comment:
for this case we may want to do an unregister (from old) and re-register (with new).